### PR TITLE
RFE 3982 Scale the Client GUI using the GUI scale

### PR DESCRIPTION
**Pull request by kuronekochomusuke**
RFE #3982 Scale the Client GUI using the GUI scale

scale the buttons, status text, panel heading, etc. on the main Client GUI using the GUI scale

image

### CODE REVIEW

**Comment by kuronekochomusuke** [comment]
i had set the window size down for a smaller picture with a really large scale.  also using the advance setting for 12 buttons per  a row, default is only 5.

here is one with more normal settings

image

**Comment by HammerGS** [comment]
Thanks! That looks really good.

**Comment by kuronekochomusuke** [comment]
also added a split pane on the the phase report ones, to make it easier to view the chat while on them.

image

image

**Comment by HammerGS** [comment]
> also added a split pane on the the phase report ones, to make it easier to view the chat while on them.
> 
> image
> 
> image

That's a nice simple QoL improvement.

**Comment by kuronekochomusuke** [comment]
this adds scaling to most of the gui using the gui scale in client settings

tried to update some areas to accommodate small and large fonts better.

these have not been updated:
- unit display -> pilot panel
- unit display -> armor panel
- lobby -> add a combat unit -> preview panel -> html display
- lobby -> view -> html display
- lobby -> set camo individual-> icon text
- lobby -> player setup -> camo -> icon text
- stuff using showInputDialog
- stuff using showMessageDialog


This is now ready for review

File: megamek/src/megamek/client/ui/dialogs/helpDialogs/AbstractHelpDialog.java [hunk 1]
```diff
@@ -69,6 +71,15 @@ protected Container createCenterPane() {
             pane.setText(Messages.getString("AbstractHelpDialog.errorReading") + e.getMessage());
             LogManager.getLogger().error("", e);
         }
-        return new JScrollPane(pane);
+
+        scrollPane = new JScrollPane(pane);
+
+        adaptToGUIScale();
+
+        return new JScrollPane(scrollPane);
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/dialogs/helpDialogs/AbstractHelpDialog.java:79]: This construct is kind of weird. Why do we have nested scrollpanes?

File: megamek/src/megamek/client/ui/swing/ForceGeneratorViewUi.java [hunk 2]
```diff
@@ -117,7 +118,7 @@ private void initUi() {
 
         forceTree = new JTree(new ForceTreeModel(null));
         forceTree.setCellRenderer(new UnitRenderer());
-        forceTree.setRowHeight(80);
+        forceTree.setRowHeight(0);
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/ForceGeneratorViewUi.java:121]: 0?

File: megamek/src/megamek/client/ui/swing/lobby/TeamOverviewPanel.java [hunk 3]
```diff
@@ -124,12 +123,6 @@ public void mouseReleased(MouseEvent e) {
         }
     };
     
-    /** Adapts the row heights and headers to the current GUI scaling. */
-    public void adaptToGUIScale() {
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/lobby/TeamOverviewPanel.java:128]: Where'd it go?

**Review by NickAragua — COMMENTED** [review]
Looks good. A few questions.

File: megamek/src/megamek/client/ui/swing/ForceGeneratorViewUi.java (hunk 2 above)
Comment (kuronekochomusuke) [reply megamek/src/megamek/client/ui/swing/ForceGeneratorViewUi.java:121]: 0 set it to auto size

Comment (NickAragua) [reply megamek/src/megamek/client/ui/swing/ForceGeneratorViewUi.java:121]: Ok, let's put in a comment explaining that.

**Review by NickAragua — APPROVED** [review]
Looks good code-wise.

**Comment by kuronekochomusuke** [comment]
let me test a bit before merge, to make sure conflicts resolved correctly.

**Pull request merged**